piet/src/util.rs

```diff
@@ -49,6 +49,11 @@ pub fn count_until_utf16(s: &str, utf16_text_position: usize) -> Option<usize> {
     None
 }
 
+/// Very naive, don't use on windows, etc etc
```

Comment:
I'm not in love with the "eof" name. Unicode uses "EOT" for this, which is less familiar but more precise.

Also I think the comment can be improved: I suggest:

```rust
/// Determine whether the string is terminated by a newline.
///
/// Note: this implementation assumes that paragraphs are separated
/// by the newline character. This will be true almost all of the time, but
/// Unicode in its infinite wisdom provides for a number of codepoints
/// that can act as paragraph separators. At some point, we should
/// probably make sure we respect those as well.
```